### ISSUE TITLE
Concatenated help strings for composite objects

### ISSUE THREAD

**Issue by maxalbert** [ioam/holoviews#330]
If I have a nested structure such as
```
:NdLayout   [Frequency,Phase]
   :HoloMap   [Exponent]
      :NdOverlay   [Shift]
         :Curve   [x]   (y)
```
and I want to find the correct object to set an option (for example `xticks`), currently I need to call `hv.help` on each of the classes `NdLayout`, `HoloMap`, `NdOverlay` and `Curve` separately and search the docstrings.

It would be great to be able to call
```
hv.help(obj, recursive=True)
```
or
```
hv.help(..., all=True)
```
or something similar. This would display the docstrings for all four objects concatenated together in a single long docstring, which would make searching for the correct option using `Ctrl+F` in the browser much less cumbersome.

**Comment by jlstevens** [comment]
Good idea. On gitter you also proposed support for regular expression search e.g ``hv.help(obj, regexp="ticks")`` which could also be very useful.

**Comment by jlstevens** [comment]
This issue has been addressed in PR #370 so I'll close the issue now. 

@maxalbert If you could try it out and see if you have any feedback, that would be great!
